modules/core/src/shadertools/src/lib/shader-module-registry.js

```diff
@@ -53,11 +52,11 @@ export default class ShaderModuleRegistry {
 
     assert(module.name, 'shader module has no name');
 
-    if (!this.shaderModules[module.name]) {
+    if (!this.shaderModules[module.name] || ignoreMultipleRegistrations) {
       module = new ShaderModule(module);
       module.dependencies = this.resolveModules(module.dependencies);
       this.shaderModules[module.name] = module;
-    } else if (!ignoreMultipleRegistrations) {
```

Comment:
Q:

I think the purpose of `ignoreMultipleRegistrations` was to prevent errors from being fired if the same module was registered twice, not to allow re-registration of modules.

Is the latter what you are trying to achieve?

Reply:
yes, i am trying to register my own picking module

Reply:
OK, maybe we should have a separate flag for that (`override: true`).

I am a little worried that the current will change to the same module being registered multiple times. But maybe it is not an issue, maybe registration is only performed once.

Reply:
there is one place in [`getShaderModule`](https://github.com/uber/luma.gl/blob/master/modules/core/src/shadertools/src/lib/shader-module-registry.js#L20-L38) where `_registerShaderModule` id called when `moduleOrName` is passed in as an object with `ignoreMultipleRegistrations: true`.